### PR TITLE
Add DoVi side data parsing

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Web**
https://github.com/jellyfin/jellyfin-web/pull/3719

**Changes**
- Add DoVi side data parsing

**Issues**
Fixes: [DoVi profile 5 in Mkv](https://www.reddit.com/r/jellyfin/comments/v2nudu/dolby_vision_to_sdr_hwa_tonemapping_is_coming/ic7a6rn/) container cannot be recognized in 10.8.0.

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -170,22 +170,24 @@ private bool IsHwTonemapAvailable(EncodingJobInfo state, EncodingOptions options
                 return isSwDecoder || isNvdecDecoder || isVaapiDecoder || isD3d11vaDecoder;
             }
 
-            return string.Equals(state.VideoStream.ColorTransfer, "smpte2084", StringComparison.OrdinalIgnoreCase)
-                   || string.Equals(state.VideoStream.ColorTransfer, "arib-std-b67", StringComparison.OrdinalIgnoreCase);
+            return string.Equals(state.VideoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase)
+                && (string.Equals(state.VideoStream.VideoRangeType, "HDR10", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(state.VideoStream.VideoRangeType, "HLG", StringComparison.OrdinalIgnoreCase));
```
Comment (cvium) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:175]:
```suggestion
            return string.Equals(state.VideoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase)
                   && (string.Equals(state.VideoStream.VideoRangeType, "HDR10", StringComparison.OrdinalIgnoreCase)
                       || string.Equals(state.VideoStream.VideoRangeType, "HLG", StringComparison.OrdinalIgnoreCase));
```

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 2]
```diff
@@ -170,22 +170,24 @@ private bool IsHwTonemapAvailable(EncodingJobInfo state, EncodingOptions options
...
                 return false;
             }
 
             // Native VPP tonemapping may come to QSV in the future.
 
-            return options.EnableVppTonemapping
-                   && string.Equals(state.VideoStream.ColorTransfer, "smpte2084", StringComparison.OrdinalIgnoreCase)
-                   && GetVideoColorBitDepth(state) == 10;
+            return string.Equals(state.VideoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase)
+                && string.Equals(state.VideoStream.VideoRangeType, "HDR10", StringComparison.OrdinalIgnoreCase);
```
Comment (cvium) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:190]:
```suggestion
            return string.Equals(state.VideoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase)
                   && string.Equals(state.VideoStream.VideoRangeType, "HDR10", StringComparison.OrdinalIgnoreCase);
```

File: MediaBrowser.MediaEncoding/Probing/MediaStreamInfo.cs [hunk 3]
```diff
@@ -310,5 +310,12 @@ public class MediaStreamInfo
         /// <value>The color primaries.</value>
         [JsonPropertyName("color_primaries")]
         public string ColorPrimaries { get; set; }
+
+        /// <summary>
+        /// Gets or sets the side_data_list.
+        /// </summary>
+        /// <value>The side_data_list.</value>
+        [JsonPropertyName("side_data_list")]
+        public IReadOnlyList<IReadOnlyDictionary<string, string>> SideDataList { get; set; }
```
Comment (cvium) [inline MediaBrowser.MediaEncoding/Probing/MediaStreamInfo.cs:319]: Why a dictionary? You could create a new class

```C#
public class MediaStreamSideData
{
    [JsonPropertyName("dv_profile")]
    public int? DvProfile { get; set; }

    ...
}
```

Comment (nyanmisaka) [reply MediaBrowser.MediaEncoding/Probing/MediaStreamInfo.cs]: Done.

File: MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        /// <value>The SideDataType.</value>
+        [JsonPropertyName("side_data_type")]
+        public string SideDataType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the DvVersionMajor.
+        /// </summary>
+        /// <value>The DvVersionMajor.</value>
+        [JsonPropertyName("dv_version_major")]
+        public int DvVersionMajor { get; set; }
```
Comment (cvium) [inline MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs:25]: Shouldnt they be nullable?

Comment (nyanmisaka) [reply MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs]: Done.

**Review by cvium — APPROVED** [review]

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs (hunk 2 above)
Comment (cvium) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:190]: Also off here

**Comment by Shadowghost** [comment]
Can you please rebase on the current release branch? You're currently reverting some changes with your PR.

**Comment by nyanmisaka** [comment]
> Can you please rebase on the current release branch? You're currently reverting some changes with your PR.

Done.

File: MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace MediaBrowser.MediaEncoding.Probing
+{
+    /// <summary>
+    /// Represents a stream within the output.
```
Comment (Shadowghost) [inline MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs:7]: Sounds a bit generic to me.

**Comment by Andy2244** [comment]
Would also love to get this fixed, so we can correctly present the DV/HDR logos for P5 content on the clients.

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**